src/FileLoaderHelper.h

```diff
@@ -171,15 +172,22 @@ inline void EnsureFileFromURLExists(std::string url, std::string file, std::stri
         }
       }
     } else {
-      // file exists but not symlink
-      printout(ERROR, "FileLoader",
-               "file " + file_path.string() + " already exists but is not a symlink");
-      printout(ERROR, "FileLoader",
-               "we tried to create a symlink " + file_path.string() + " to the actual resource, " +
-                   "but a file already exists there and we will not remove it automatically");
-      printout(ERROR, "FileLoader", "hint: backup the file, remove it manually, and retry");
-      std::_Exit(EXIT_FAILURE);
+      // file does not exists, i.e. dead symllink -> remove symlink
+      if (fs::remove(file_path) == false) {
+        // failure mode; cannot remove dead symlink
+        printout(ERROR, "FileLoader", "unable to remove symlink " + file_path.string());
+        std::_Exit(EXIT_FAILURE);
+      }
     }
+  } else {
+    // failure mode: file exists but not symlink, and we won't remove files
+    printout(ERROR, "FileLoader",
+             "file " + file_path.string() + " already exists but is not a symlink");
+    printout(ERROR, "FileLoader",
+             "we tried to create a symlink " + file_path.string() + " to the actual resource, " +
+                 "but a file already exists there and we will not remove it automatically");
+    printout(ERROR, "FileLoader", "hint: backup the file, remove it manually, and retry");
+    std::_Exit(EXIT_FAILURE);
```

Comment:
```suggestion
    if (fs::exists(file_path)) {
      // failure mode: file exists but not symlink, and we won't remove files
      printout(ERROR, "FileLoader",
               "file " + file_path.string() + " already exists but is not a symlink");
      printout(ERROR, "FileLoader",
               "we tried to create a symlink " + file_path.string() + " to the actual resource, " +
                   "but a file already exists there and we will not remove it automatically");
      printout(ERROR, "FileLoader", "hint: backup the file, remove it manually, and retry");
      std::_Exit(EXIT_FAILURE);
    }
```

Reply:
I'll look at this tomorrow. And maybe write a ci job just to test this.

Reply:
The CI agrees with me.

Reply:
In this case in particular: just because it compiles doesn't mean this will now work.